include/DataProvider.hpp

```diff
@@ -0,0 +1,46 @@
+/*
+*   DataProvider.hpp
+*   ----------------------
+*   Created on: 2023/09/29
+*   Author: Lankow
+*/
+#ifndef DATA_PROVIDER_HPP
+#define DATA_PROVIDER_HPP
+
+#include <Arduino.h>
+#include "constants.hpp"
+
+class DataProvider
+{
+    private:
+    uint16_t m_currentHumidityLvl[MAX_SENSORS_NO];
+    uint16_t m_humidityThreshold[MAX_SENSORS_NO];
+    bool m_humidityActive[MAX_SENSORS_NO];
+    uint8_t m_sensorToWater;
+
+    char m_currentTime[TIMESTAMP_LENGTH];
+    float m_temperature;
+    float m_roomHumidity;
+
+    public:
+    DataProvider();
+
+    float& getTemperature();
+    float& getRoomHumidity();
+    char* getCurrentTime();
+    uint8_t& getSensorToWater();
+    uint16_t* getCurrentHumidityLvl();
```

Comment:
Oh boy. Zarządzanie tablicami jako wskaźnikami to zło. 
Zamiast trzymać to w zwykłych tablicach, wykorzystaj std::vector albo std::array

Reply:
Dlaczego zło?